scripts/globals/limbus.lua

```diff
@@ -493,3 +482,189 @@ function xi.limbus.spawnRandomCrate(npc, battlefield, var, mask, canMimic)
         return
     end
 end
+
+function xi.limbus.showCrate(crateID)
+    local crate = GetNPCByID(crateID)
+    crate:setStatus(xi.status.NORMAL)
+    crate:setUntargetable(false)
+    crate:resetLocalVars()
+end
+
+function xi.limbus.showRecoverCrate(crateID)
+    local crate = GetMobByID(crateID)
+    crate:setAnimationSub(8)
+    crate:setStatus(xi.status.NORMAL)
+    crate:setUntargetable(false)
+    crate:resetLocalVars()
+end
+
+function xi.limbus.hideCrate(crate)
+    crate:setStatus(xi.status.DISAPPEAR)
+    crate:setUntargetable(true)
+    crate:resetLocalVars()
+end
+
+function xi.limbus.spawnFrom(mob, crateID)
+    local crate = GetNPCByID(crateID)
+    crate:setPos(mob:getXPos(), mob:getYPos(), mob:getZPos(), mob:getRotPos())
+    crate:setStatus(xi.status.NORMAL)
+    crate:setUntargetable(false)
+    crate:AnimationSub(8)
+end
+
+function xi.limbus.spawnRecoverFrom(mob, crateID)
+    local crate = GetMobByID(crateID)
+    crate:setPos(mob:getXPos(), mob:getYPos(), mob:getZPos(), mob:getRotPos())
+    xi.limbus.showRecoverCrate(crateID)
+end
+
+function xi.limbus.openDoor(battlefield, doorID)
+    local door = GetNPCByID(doorID)
+    local ID = zones[door:getZoneID()]
+    local remaining = battlefield:getRemainingTime() / 60
+
+    for i, player in pairs(battlefield:getPlayers()) do
+        player:messageSpecial(ID.text.GATE_OPEN)
+        player:messageSpecial(ID.text.TIME_LEFT, remaining)
+    end
+    door:setAnimation(xi.animation.OPEN_DOOR)
+end
+
+function xi.limbus.extendTimeLimit(ID, battlefield, amount)
+    local timeLimit = battlefield:getTimeLimit()
+    battlefield:setTimeLimit(timeLimit + amount * 60)
+    local remaining = battlefield:getRemainingTime() / 60
+
+    for _, player in pairs(battlefield:getPlayers()) do
+        player:messageSpecial(ID.text.TIME_EXTENDED, amount)
+        player:messageSpecial(ID.text.TIME_LEFT, remaining)
+    end
+end
+
+LimbusArea = {}
+LimbusArea.__index = LimbusArea
+LimbusArea.__eq = function(m1, m2)
+    return m1.name == m2.name
+end
+
+LimbusArea.name = ""
+LimbusArea.serverVar = ""
+LimbusArea.requiredCard = 0
+LimbusArea.groups = {}
+LimbusArea.paths = {}
+LimbusArea.loot = {}
+
+function LimbusArea:new(name, requiredCard)
```

Comment:
I like the direction this is going, and would like to see this eventually move towards a tabled format similar to how interaction framework is handled.  Perhaps as well, since this seems to wrap battlefields, a more generic base class could be made to implement this design, and then have limbus extend it (similar to the Container class which is utilized by Mission and Quest)

Reply:
I'd be happy to chat with you about this more. Hit me up on discord when you have time. What we have now could work for this PR and then I can expand upon it for a separate PR.